go-controller/pkg/ovn/port_cache.go

```diff
@@ -29,21 +33,49 @@ type lpInfo struct {
 func newPortCache(stopChan <-chan struct{}) *portCache {
 	return &portCache{
 		stopChan: stopChan,
-		cache:    make(map[string]*lpInfo),
+		cache:    make(map[string]map[string]*lpInfo),
+	}
+}
+
+func (c *portCache) get(pod *kapi.Pod, nadName string) (*lpInfo, error) {
+	var logicalPort string
+
+	podName := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+	if nadName == types.DefaultNetworkName {
+		logicalPort = util.GetLogicalPortName(pod.Namespace, pod.Name)
+	} else {
+		logicalPort = util.GetSecondaryNetworkLogicalPortName(pod.Namespace, pod.Name, nadName)
+	}
+	c.RLock()
+	defer c.RUnlock()
+	if infoMap, ok := c.cache[podName]; ok {
+		if info, ok := infoMap[nadName]; ok {
+			x := *info
+			return &x, nil
+		}
 	}
+	return nil, fmt.Errorf("logical port %s for pod %s not found in cache", podName, logicalPort)
 }
 
-func (c *portCache) get(logicalPort string) (*lpInfo, error) {
+func (c *portCache) getAll(pod *kapi.Pod) (map[string]*lpInfo, error) {
+	podName := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
 	c.RLock()
 	defer c.RUnlock()
-	if info, ok := c.cache[logicalPort]; ok {
-		x := *info
-		return &x, nil
+	if infoMap, ok := c.cache[podName]; ok {
+		return infoMap, nil
```

Comment:
unsafe return. You are returning the map here directly instead of a copy of it after you unlock the cache.

Reply:
again, thanks for caching this!